zebra-state/src/service.rs

```diff
@@ -491,6 +485,10 @@ impl StateService {
             }
 
             self.drain_queue_and_commit_finalized();
+
+            if let Some(finalized_tip_height) = self.read_service.db.finalized_tip_height() {
```

Comment:
Is this code move or extra code needed?

Reply:
Do we also need to clear the queue in the "finished finalized blocks" case?

Reply:
> Is this code move or extra code needed?

It's needed to prune the finalized block hashes that are added every time a block is sent to the write task, it was only being pruned when non-finalized block commits were queued.

> Do we also need to clear the queue in the "finished finalized blocks" case?

I think it's okay to leave them there until the finalized tip grows past their heights, and the state needs UTXOs from the finalized blocks near the last checkpoint to verify the first non-finalized blocks.